### PR TITLE
Test basic S3 I.O, object listing and data integrity, add the S3Client class and make some necessary account creation adustments

### PR DESCRIPTION
**Pull request by sagihirshfeld**
### Explain the changes
1. Add `extended_connection.py` and the `ExtendedConnection` class. This class inherits the [base Connection class](https://github.com/red-hat-storage/common-ci-utils/blob/main/common_ci_utils/connection.py) that was used in SSHConnection, and extends it by adding a `download_file` method which was necessary in order to download the TLS certificate from the remote NSFS server to the local machine.
 
2. Rename `SSHConnection` to `SSHConnectionManager`, and the name of its containing file from `connection.py` to `ssh_connection_manager`. The main reason this was necessary was distinguish it from the `extended_connections.py` file and the `ExtendedConnection` class.

3. Add the `setup_nsfs_server_tls_certificate` fixture to `conftest.py`. This fixture sets up up the NSFS TLS key and certificate on the remote NSFS server according to [the documented procedure](https://github.com/noobaa/noobaa-core/blob/6afbd87c2c79f0b54a9ccd1784ef27465214839c/docs/non_containerized_NSFS.md?plain=1#L470-L549).

4. Add the S3Client class and s3_client_factory fixture: this class is used to store the necessary credentials for making S3 commands, and serves as a wrapper for making S3 commands via boto3 and the AWS S3 CLI.

5. Add the tmp_directories_factory fixture. This fixture allows setting up temporary directories under `/tmp` that will be deleted during teardown.

6. Add `test_basic_s3.py`. `test_basic_s3` checks the following:
- Create an account and a bucket
- Write objects to the bucket
- List the bucket's contents
- Read the objects from the bucket and verify data integrity
- Delete the objects from the bucket

7. Adjust account creation to latest noobaa-core changes

8. Some code formatting that was enforced by Black, as can be seen in `bucket.py`

### CODE REVIEW

File: noobaa_sa/account.py [hunk 1]
```diff
@@ -79,9 +91,11 @@ def create(self, account_name, access_key, secret_key, config_root=None):
         account_data = {
             "account_name": account_name,
             "account_email": account_email,
+            "creation_date": datetime.now().strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
```
Comment (udaysk23) [inline noobaa_sa/account.py:94]: We don't need creation date since node CLI is creating that value dynamically while executing the command

Comment (sagihirshfeld) [reply noobaa_sa/account.py:94]: I was getting the following error without it:
```
E           noobaa_sa.exceptions.AccountCreationFailed: Creation of account failed with error {
E             "error": {
E               "code": "InvalidSchema",
E               "message": "Schema invalid, please use required properties",
E               "detail": "Error: must have required property 'creation_date'\n    at Object.validate_account_schema (/usr/local/noobaa-core/src/manage_nsfs/nsfs_schema_utils.js:37:23)\n    at add_account (/usr/local/noobaa-core/src/cmd/manage_nsfs.js:526:23)\n    at async manage_account_operations (/usr/local/noobaa-core/src/cmd/manage_nsfs.js:619:9)\n    at async account_management (/usr/local/noobaa-core/src/cmd/manage_nsfs.js:391:5)\n    at async main (/usr/local/noobaa-core/src/cmd/manage_nsfs.js:191:13)"
E             }
E           }
```

I'll check with the devs if that's the expected behavior

Comment (udaysk23) [reply noobaa_sa/account.py:94]: On latest noobaa build, it got fixed and it uses date dynamically

Comment (sagihirshfeld) [reply noobaa_sa/account.py:94]: I'm not sure if it was fixed at one point, but after confirming with Romy it was determined it's a BZ on the latest build: https://bugzilla.redhat.com/show_bug.cgi?id=2260325

**Comment by sagihirshfeld** [comment]
Closing in favor of #8 and #11 for the sake of modularity

**Pull request closed without merging**